pymod/ams.py

```diff
@@ -53,6 +59,28 @@ def __init__(self):
                              "auth_x509": ["post", set([400, 401, 403, 404])],
                              "sub_pull": ["post", set([400, 401, 403, 404])]}
 
+    def _retry_make_request(self, url, body=None, route_name=None, retry=3, retrysleep=60, **reqkwargs):
+        i = 1
+        timeout = reqkwargs.get('timeout', 0)
+
+        try:
+            while i <= retry + 1:
+                try:
+                    return self._make_request(url, body, route_name, **reqkwargs)
+                except AmsConnectionException as e:
+                    if i == retry + 1:
+                        raise e
+                    else:
+                        time.sleep(retrysleep)
+                        if timeout:
+                            log.warning('Retry #{0} after {1} seconds, connection timeout set to {2} seconds'.format(i, retrysleep, timeout))
+                        else:
+                            log.warning('Retry #{0} after {1} seconds'.format(i, retrysleep))
+                finally:
+                    i += 1
+        except AmsConnectionException as e:
+            raise e
```

Comment:
If you're only raising the exception you're catching, then the `try...except` is redundant.

Reply:
Outer `try...except` clause was just a placeholder for something that I thought I would use, but I think I will not in the end so it's removed.

Reply:
Fair enough. :)